user_profile/views.py

```diff
@@ -0,0 +1,268 @@
+
+
+class UserFeedView(TemplateView, LoginRequiredMixin):
+    """
+    User Feed View that renders the user feed
+    Only supports get request
+    takes the user id as argument and return and renders
+    feed objects for that user
+    ]
+    """
+    login_url = 'user_profile:index'
+    template_name = 'pages/user_feed.html'
+
+    def get(self, request, user_id, *args, **kwargs):
+        if user_id == request.user.id:
+            return redirect('user_profile:settings')
+        context = super(UserFeedView, self).get_context_data(*args, **kwargs)
+        try:
+            context['feed_user'] = CustomUser.objects.get(id=user_id)
+            if context['feed_user'].is_staff or context['feed_user'].is_superuser:
+                raise CustomUser.DoesNotExist
+        except CustomUser.DoesNotExist:
+            messages.error(
+                request, "User with user_id={} doesnt Exist!".format(user_id))
+            return redirect('user_profile:settings')
+        try:
+            context['feed_objects'] = models.Feed.objects.filter(
+                user_id=user_id)
+        except Exception:
+            messages.error(request, "Error Loading Feed for this user")
+        return render(request, template_name=self.template_name, context=context)
+
+
+class SearchView(ListView, LoginRequiredMixin):
+    """
+    User Search View
+    Inherits from ListView and LoginRequiredMixin
+    supports both get and post request
+    the post request is an ajax call
+    """
+    login_url = 'user_profile:index'
+    template_name = 'pages/search.html'
+    model = models.UserProfile
+
+    def get_queryset(self):
+        """
+        Functions filters the search objects
+        only shows users that match the search critera
+        either first name or last name should contain the search keyword
+        query is further filtered for users that are not staff or admin
+        also the current user is filtered out of the results
+        """
+        qs = super().get_queryset()
+        # filter to get users that match our criteria
+        query = self.request.GET['search_query'] if self.request.GET.get(
+            'search_query', None) else self.request.POST['search_query']  # in case of ajax request post contains query else get contains
+        qs = qs.filter(Q(user__first_name__icontains=query)
+                       | Q(user__last_name__icontains=query))
+        qs = qs.filter(~Q(user__is_staff=True) and ~Q(
+            user__is_superuser=True), ~Q(user__id=self.request.user.id))  # filter non admin users out and remove current logged in user
+        return qs
+
+    def get_context_data(self, **kwargs):
+        """
+        Method to add additional context data
+        for the side filters
+        we only wanna include values that are present in the search results
+        to be present in the filters hence these filtering
+        """
+        queryset = self.get_queryset().values_list(
```

Comment:
you can use <field_name>__isnull=False to filter out null vlaues.

Reply:
fixed

Reply:
also filtered these using list comprehensions